polaris/ocean/convergence/analysis.py

```diff
@@ -395,11 +395,18 @@ def get_output_field(self, refinement_factor, field_name, time, zidx=None):
         field_mpas : xarray.DataArray
             model output field
         """
-        ds_out = self.open_model_dataset(f'output_r{refinement_factor:02g}.nc')
+        config = self.config
+        ds_out = self.open_model_dataset(f'output_r{refinement_factor:02g}.nc',
+                                         decode_times=False)
 
-        tidx = time_index_from_xtime(ds_out.xtime.values, time)
-        ds_out = ds_out.isel(Time=tidx)
+        model = config.get('ocean', 'model')
+        if model == 'mpas-o':
+            tidx = time_index_from_xtime(ds_out.xtime.values, time)
+        else:
+            # time is seconds since the start of the simulation in Omega
+            tidx = np.argmin(np.abs(ds_out.Time.values - time))
```

Comment:
```suggestion
        if model == 'mpas-o':
            dt = time_since_start(ds_out.xtime.values)
        else:
            # time is seconds since the start of the simulation in Omega
            dt = ds_out.Time.values
tidx = np.argmin(np.abs(dt - time))
```

Can we break up `time_index_from_xtime` into two functions: 
```
def time_index_from_xtime(xtime, dt_target, start_xtime=None):
    dt = time_since_start(ds_out.xtime.values, start_xtime=None)
    time_index = np.argmin(np.abs(np.subtract(dt, dt_target)))
    return time_index

def time_since_start(xtime, start_xtime=None):
    if start_xtime is None:
        start_xtime = xtime[0].decode()

    t0 = datetime.datetime.strptime(start_xtime, '%Y-%m-%d_%H:%M:%S')
    dt = np.zeros((len(xtime),))
    for idx, xt in enumerate(xtime):
        t = datetime.datetime.strptime(xt.decode(),
                                       '%Y-%m-%d_%H:%M:%S')
        dt[idx] = (t - t0).total_seconds()
        return dt

Reply:
Sounds good!